samples/web/content/apprtc/apprtc.py

```diff
@@ -495,12 +515,109 @@ def write_response(self, result, params, messages):
       'result': result,
       'params': params
     }))
+    
+  def report_error(self, error):
+    self.write_response(error, {}, [])
 
   def write_room_parameters(self, room_id, client_id, messages, is_initiator):
     params = get_room_parameters(self.request, room_id, client_id, is_initiator)
     self.write_response('SUCCESS', params, messages)
+    
+  def handle_call(self, msg, room_id):
```

Comment:
Does it make more sense for caller to call with no room id, in which case the server assigns and returns an unused one? (For those cases where collisions happen this avoids an additional request). Callee still needs to accept by joining that particular room though. Alternately I suppose caller can call with a uuid.